installinator/src/peers.rs

```diff
@@ -135,6 +134,10 @@ impl FetchedArtifact {
                         "(attempt {attempt}) failed to discover peers, retrying: {}",
                         DisplayErrorChain::new(AsRef::<dyn std::error::Error>::as_ref(&error)),
                     );
+                    cx.send_progress(StepProgress::retry(format!(
+                        "failed to discover peers: {error}"
+                    )))
+                    .await;
```

Comment:
If we haven't found any peers, this retry message ultimately gets dropped, right?

Reply:
Yes, it will be buffered up, and then dropped if we run out of capacity.